src/humpty_error.rs

```diff
@@ -0,0 +1,116 @@
+//! Error stuff.
+//! TODO docs before release
+#![allow(missing_docs)]
+
+use crate::http::method::Method;
+use crate::http::request::HttpVersion;
+use std::error::Error;
+use std::fmt::{Debug, Display, Formatter};
+use std::io;
+use std::io::ErrorKind;
+
+pub type HumptyResult<T> = Result<T, HumptyError>;
+
+#[derive(Debug)]
+pub enum RequestHeadParsingError {
```

Comment:
This should probably have `#[non_exhaustive]`

Reply:
done